Mathlib/LinearAlgebra/Span.lean

```diff
@@ -92,20 +92,22 @@ theorem span_coe_eq_restrictScalars [Semiring S] [SMul S R] [Module S M] [IsScal
   span_eq (p.restrictScalars S)
 #align submodule.span_coe_eq_restrict_scalars Submodule.span_coe_eq_restrictScalars
 
+theorem image_span_subset (f : F) (s : Set M) (N : Submodule R₂ M₂) :
+    f '' span R s ⊆ N ↔ ∀ m ∈ s, f m ∈ N := image_subset_iff.trans <| span_le (p := N.comap f)
+
+theorem image_span_subset_span (f : F) (s : Set M) : f '' span R s ⊆ span R₂ (f '' s) :=
+  (image_span_subset f s _).2 fun x hx ↦ subset_span ⟨x, hx, rfl⟩
```

Comment:
```suggestion
/-- A version of `Submodule.RingHomSurjective` that does not require the `RingHomSurjective`
assumption. -/
theorem image_span_le (f : F) (s : Set M) (N : Submodule R₂ M₂) :
    f '' span R s ≤ N ↔ MapsTo f s N := image_subset_iff.trans <| span_le (p := N.comap f)

theorem image_span_le_span (f : F) (s : Set M) : f '' span R s ≤ span R₂ (f '' s) :=
  (image_span_le f s _).2 fun x hx ↦ subset_span ⟨x, hx, rfl⟩
```

Reply:
Oh, and these are sets not submodules, so `⊆` should be used instead of `≤` and the name should stay the same.

Reply:
The issue is that if the RingHom isn't surjective then the image is not necessarily a submodule.

Reply:
Yes I realise this!

FWIW here's what happened at my end:
 1. These are submodules, they should use `≤`, I'll suggest that!
 2. Wait, shouldn't these be stated with `Submodule.map`? Oh I see: without the required surjectivity this is a statement about sets. This is slightly subtle so I'll suggest adding a comment.
 
except that I missed out on:
 3. Realise that step 1 no longer applies, so I should remove my suggestion from step 1.